sw/device/tests/rv_core_ibex_address_translation_test.c

```diff
@@ -0,0 +1,204 @@
+      .remap_addr = make_lower_case_addr,
+      .size = next_power_of_two(kMakeLowerCaseFnSize),
+  };
+  dif_rv_core_ibex_addr_translation_mapping_t get_name_mapping = {
+      .matching_addr = kRemapAddr,
+      .remap_addr = get_name_addr,
+      .size = next_power_of_two(kGetNameFnSize),
+  };
+  const dif_rv_core_ibex_addr_translation_bus_t kBuses[] = {
+      kDifRvCoreIbexAddrTranslationIBus, kDifRvCoreIbexAddrTranslationDBus};
+  const dif_rv_core_ibex_addr_translation_slot_t kSlots[] = {
+      kDifRvCoreIbexAddrTranslationSlot_0, kDifRvCoreIbexAddrTranslationSlot_1};
+
+  // Get ibex core handle.
+  dif_rv_core_ibex_t ibex_core;
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &ibex_core));
+
+  // Map virtual address space to make_lower_case() using slot 1.
+  for (size_t idx = 0; idx < 2; ++idx) {
+    CHECK_DIF_OK(dif_rv_core_ibex_configure_addr_translation(
+        &ibex_core, kSlots[1], kBuses[idx], make_lower_case_mapping));
+  };
+
+  // Enable address translation slot 1.
+  for (size_t idx = 0; idx < 2; ++idx) {
+    CHECK_DIF_OK(dif_rv_core_ibex_enable_addr_translation(&ibex_core, kSlots[1],
+                                                          kBuses[idx]));
+  };
+
+  // Reset test string content.
+  memcpy(test_str, TEST_STR, sizeof(test_str));
+
+  // Run make_lower_case() from virtual memory and check the result.
+  ((str_fn_t)kRemapAddr)(test_str);
+  CHECK_STR_EQ(test_str, EXPECTED_RESULT_MAKE_LOWER_CASE);
+
+  // Remap virtual address space to get_name() using slot 1.
+  for (size_t idx = 0; idx < 2; ++idx) {
+    CHECK_DIF_OK(dif_rv_core_ibex_configure_addr_translation(
+        &ibex_core, kSlots[1], kBuses[idx], get_name_mapping));
+  };
+
+  // Run get_name() from virtual memory and check the result.
+  ((str_fn_t)kRemapAddr)(test_str);
+  CHECK_STR_EQ(test_str, EXPECTED_RESULT_GET_NAME);
+
+  /////////////////////////////////////////////////////////////////////////////
+  // Check slot 0 has higher priority than slot 1.
+  /////////////////////////////////////////////////////////////////////////////
+  //
+  // Map virtual address space to make_lower_case() but using slot 0.
+  for (size_t idx = 0; idx < 2; ++idx) {
+    CHECK_DIF_OK(dif_rv_core_ibex_configure_addr_translation(
+        &ibex_core, kSlots[0], kBuses[idx], make_lower_case_mapping));
+  };
+
+  // Enable address translation slot 0.
+  for (size_t idx = 0; idx < 2; ++idx) {
+    CHECK_DIF_OK(dif_rv_core_ibex_enable_addr_translation(&ibex_core, kSlots[0],
+                                                          kBuses[idx]));
+  };
+
+  // Reset test string content.
+  memcpy(test_str, TEST_STR, sizeof(test_str));
+
+  // Run get_name() from virtual memory and check the result.
+  ((str_fn_t)kRemapAddr)(test_str);
+  CHECK_STR_EQ(test_str, EXPECTED_RESULT_MAKE_LOWER_CASE);
+
+  // TODO: Turn off address translation and confirm regions are no longer being
```

Comment:
Sorry to bing that know, but I'm came up with a new idea, what if you finish the test which will make it fail, then you make this test as broken in the bazel file with this TODO?

Reply:
I agree, we have some other tests with failures that we mark as broken, with a TODO indicating the issue.

Reply:
Disable test is added back in and the test is tagged as 'broken' in the BUILD file.